src/App.jsx

```diff
@@ -1,17 +1,40 @@
+import useSWR from "swr";
 import { useState } from "react";
 import "./App.css";
 
 function App() {
   const url = "https://httpstat.us/200?sleep=2000";
   const headers = { Accept: "application/json" };
 
-  const [status, setStatus] = useState("");
+    const fetcher = (url) => fetch(url, { headers }).then((res) => res.json());
 
-  fetch(url, { headers })
-    .then((res) => res.json())
-    .then((json) => setStatus(json.description));
+    const { data, error, isLoading, mutate } = useSWR(url, fetcher);
+    const [isLoadingButton, setIsLoadingButton] = useState(false);
+    const [status, setStatus] = useState("");
 
-  return <>{status && <p>Status : {status}</p>}</>;
+    const handleLoad = async () => {
+        setIsLoadingButton(true);
+        setStatus("Loading...");
+        await mutate();
+        setIsLoadingButton(false);
+        setStatus(data ? data.description : "OK");
+    };
+
+    if (error) return <p className="error">Failed to load.</p>;
+
+    return (
+        <div className="app">
+            <h1>Status Checker</h1>
+            <div className="status-box">
+                <p>Status: {isLoadingButton || isLoading ? "Loading..." : status || "OK"}</p>
+            </div>
+            {!isLoadingButton && !isLoading && (
+                <button onClick={handleLoad} className="load-button">
```

Comment:
```suggestion
                <button onClick={mutate} className="load-button">
```

ここも state が isLoading とかになればこんな感じでシンプルに mutate させるだけにできそう？

※ mutate だと isLoading が変わらないかも。 revalidate とかする必要があるかもしれません